packages/client/src/features/game/cards-game/cards-game.ts

```diff
@@ -0,0 +1,85 @@
+import { Dispatch, SetStateAction } from 'react'
+import { Card } from '../../../shared/card/Card'
+import { TCard, TRect } from '../types'
+import { closedCardImage } from '../assets'
+import imports from '../imports'
+
+export const CardsGame = (
+  ctx: CanvasRenderingContext2D,
+  widthGame: number,
+  heightGame: number,
+  isPlayerCards: boolean,
+  cardsArray: TCard[],
+  setSelectedSrcCardToMove: Dispatch<SetStateAction<string>>,
+  isMovePlayer: boolean,
+) => {
+  const x = Math.round(
+    (widthGame - cardsArray.length * (imports.CARD_WIDTH + 15)) / 2,
+  )
+  const yBot = 25
+  const yPlayer = Math.round(heightGame - imports.CARD_HEIGHT - 45)
+
+  const canvas = document.querySelector('canvas')
+
+  const newCardsArray = cardsArray.map(item =>
+    isPlayerCards ? item.image : closedCardImage,
+  )
+
+  const sprites = imports.spritesLoaded(newCardsArray)
+
+  Promise.all(sprites)
+    .then(images => {
+      const rects: TRect[] = []
+      let newX = x
+
+      images.forEach(item => {
+        rects.push({ image: item, x: newX, y: isPlayerCards ? yPlayer : yBot })
+        newX = newX + imports.CARD_WIDTH + 15
+      })
+
+      const wait = imports.debounce(() => {
+        ctx.clearRect(0, isPlayerCards ? yPlayer : yBot, widthGame, 200)
+        rects.forEach(item => {
+          Card(ctx, false, item.image, item.x, item.y, false)
+        })
+      }, 50)
+      wait()
+
+      if (isPlayerCards) {
+        if (canvas) {
```

Comment:
Похоже, что эти два ифа можно объединить
```suggestion
      if (isPlayerCards && canvas) {
```